pinot-spi/src/main/java/org/apache/pinot/spi/config/table/PreAggregationConfig.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.table;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.pinot.spi.config.BaseJsonConfig;
+
+
+public class PreAggregationConfig extends BaseJsonConfig {
```

Comment:
It would be more flexible if we take a function expression and a destination column. Check `TransformConfig` as an example. This way we can handle aggregation with multiple arguments (e.g. `DistinctCountHLL` with custom `log2m`). Conceptually we can even handle nested expression such as `sum(colA - colB)`.

To make this config consistent with `TransformConfig`, we can name the fields `_columnName` and `preAggregationFunction`

Reply:
Updated!

Reply:
What I meant is to keep the following 2 fields:
`_columnName`: the dest column name
`_preAggregationFunction`: the whole function expression instead of just the function type, e.g. `SUM(c1)`

Reply:
Thanks for this suggestion. I hadn't thought about doing it this way, but I agree that it will add more flexibility later on.